src/Diagnostics/CodeAnalysis/Core/CodeAnalysisDiagnosticsResources.resx

```diff
@@ -174,6 +174,26 @@
   <data name="InvalidSyntaxKindTypeArgumentDescription" xml:space="preserve">
     <value>DiagnosticAnalyzer's language-specific Register methods, such as RegisterSyntaxNodeAction, RegisterCodeBlockStartAction and RegisterCodeBlockEndAction, expect a language-specific 'SyntaxKind' type argument for it's '{0}' type parameter. Otherwise, the registered analyzer action can never be invoked during analysis.</value>
   </data>
+  <data name="StartActionWithOnlyEndActionTitle" xml:space="preserve">
+    <value>Start action has no registered non-end actions.</value>
+  </data>
+  <data name="StartActionWithNoRegisteredActionsMessage" xml:space="preserve">
+    <value>You did not register any analyzer action on '{0}'. Move all the action registrations in '{1}' that depend on this start action onto '{0}'.</value>
+  </data>
+  <data name="StartActionWithNoRegisteredActionsTitle" xml:space="preserve">
+    <value>Start action has no registered actions.</value>
+  </data>
+  <data name="StartActionWithOnlyEndActionMessage" xml:space="preserve">
+    <value>You did not register any analyzer action on '{0}', except for a '{1}'. Either replace this start/end action pair with a '{2}' or move all the action registrations in '{3}' that depend on this start action onto '{0}'.</value>
+  </data>
+  <data name="StatefulAnalyzerRegisterActionsDescription" xml:space="preserve">
+    <value>An analyzer start action enables performing stateful analysis over a given code unit, such as a code block, compilation, etc in a performant way. However, incorrect design for such analyzers can lead to memory leaks and non-performant analyzer execution. Use the following guidelines for writing such analyzers:
+1. Define a new scope for the registered start action, possibly with a private nested type for analyzing each code unit.
+2. If required, define and intialize state object(s) in the start action within this scope.
```

Comment:
"initialize" is misspelled. "state objects(s)" can probably be shortened to "state". "within this scope" is not necessary.